src/components/Output/custom-els/TwoUp/index.ts

```diff
@@ -52,39 +53,64 @@ export default class TwoUp extends HTMLElement {
     });
   }
 
-  connectedCallback () {
+  connectedCallback() {
     this._childrenChange();
     if (!this._everConnected) {
-      // Set the initial position of the handle.
-      requestAnimationFrame(() => {
-        const bounds = this.getBoundingClientRect();
-        this._position = bounds.width / 2;
-        this._setPosition();
-      });
+      this._resetPosition();
       this._everConnected = true;
     }
   }
 
+  attributeChangedCallback(name: string) {
+    if (name === verticalAttr) {
+      this._resetPosition();
+    }
+  }
+
+  private _resetPosition() {
+    // Set the initial position of the handle.
+    requestAnimationFrame(() => {
+      const bounds = this.getBoundingClientRect();
+      this._position = (this.vertical ? bounds.height : bounds.width) / 2;
+      this._setPosition();
+    });
+  }
+
   /**
    * If true, this element works in browsers that don't support clip-path (Edge).
    * However, this means you'll have to set the height of this element manually.
    */
-  get noClipPathCompat () {
-    return this.hasAttribute(legacyClipCompat);
+  get legacyClipCompat() {
+    return this.hasAttribute(legacyClipCompatAttr);
   }
 
-  set noClipPathCompat (val: boolean) {
+  set legacyClipCompat(val: boolean) {
     if (val) {
-      this.setAttribute(legacyClipCompat, '');
+      this.setAttribute(legacyClipCompatAttr, '');
     } else {
-      this.removeAttribute(legacyClipCompat);
+      this.removeAttribute(legacyClipCompatAttr);
+    }
+  }
+
+  /**
+   * Split vertically rather than horizontally.
+   */
+  get vertical() {
+    return this.hasAttribute(verticalAttr);
+  }
```

Comment:
There's repetition here, but I'm not sure if there's a typescript-friendly way to solve it.

I'd usually transform the attr names to property names then generate the getters/setters automatically.

Reply:
We could write a decorator for tying attrs with props. Not sure if it’s worth it (yet).

Reply:
If we feel we need it, we should talk to the polymer folks. Pretty sure they'll have thought about this before.